publication-model/src/test/java/no/unit/nva/model/EntityDescriptionTest.java

```diff
@@ -193,6 +260,145 @@ void shouldProvideContributorSequenceAndKeepStableSorting() {
+        var reference = entityDescription.getReference();
+        reference.setPublicationContext(createContextForType(reference, publicationChannelYear));
+        entityDescription.setPublicationDate(publicationDateWithYearOnly(publicationYear));
+        return entityDescription;
+    }
+
+    private static PublicationContext createContextForType(Reference reference, String publicationChannelYear) {
+        if (isAssignableFrom(reference, Book.class)) {
+            return createBookSubclassWithSeriesAndPublisher((Book) reference.getPublicationContext(), publicationChannelYear);
+        } else if (isAssignableFrom(reference, Degree.class)) {
+            return createBookSubclassWithSeriesAndPublisher((Degree) reference.getPublicationContext(), publicationChannelYear);
+        } else if (isAssignableFrom(reference, GeographicalContent.class)) {
+            return new GeographicalContent(publisher(publicationChannelYear));
+        } else if (isAssignableFrom(reference, Journal.class)) {
+            return new Journal(serialPublicationUri(publicationChannelYear));
+        } else if (isAssignableFrom(reference, Report.class)) {
+            return  createBookSubclassWithSeriesAndPublisher((Report) reference.getPublicationContext(), publicationChannelYear);
+        } else if (isAssignableFrom(reference, ResearchData.class)) {
+            return new ResearchData(publisher(publicationChannelYear));
+        } else {
+            throw new IllegalStateException("Unexpected value: " + reference.getPublicationContext());
+        }
+    }
+
+    private static Book createBookSubclassWithSeriesAndPublisher(Book book, String publicationYear) {
+        return book.copy()
+                .withSeries(seriesUri(publicationYear)).withPublisher(publisher(publicationYear)).build();
+    }
+
+    private static boolean isAssignableFrom(Reference reference, Class<? extends PublicationContext> clazz) {
+        return Objects.requireNonNull(reference.getPublicationContext()).getClass().isAssignableFrom(clazz);
+    }
+
+    private static EntityDescription entityDescriptionFromContext(Class<? extends PublicationContext> clazz) {
+        if (clazz.equals(Book.class)) {
+            return randomEntityDescription(AcademicMonograph.class);
+        } else if (clazz.equals(Degree.class)) {
+            return randomEntityDescription(DegreePhd.class);
+        } else if (clazz.equals(GeographicalContent.class)) {
+            return randomEntityDescription(Map.class);
+        } else if (clazz.equals(Journal.class)) {
+            return randomEntityDescription(AcademicArticle.class);
+        } else if (clazz.equals(Report.class)) {
+            return randomEntityDescription(ReportResearch.class);
+        } else if (clazz.equals(ResearchData.class)) {
+            return randomEntityDescription(DataSet.class);
+        } else {
+            throw new IllegalStateException("Unexpected value: " + clazz);
+        }
+    }
```

Comment:
Same question here: Is `Degree` and `Report` reachable, since they extend `Book`?

Reply:
Yes, the matching is strict equality, rather than being an instance (or successor) of